pkg/config/config.go

```diff
@@ -76,57 +105,128 @@ func SaveToFile(config *Config, filename string) error {
+			//return fmt.Errorf("wrong value specified for 'who_can_contact_owner' field (group: %s)", group.Name)
+			allTheErrors = append(allTheErrors, fmt.Errorf("wrong value specified for 'who_can_contact_owner' field (group: %s)", group.Name))
+		}
+
+		if !(strings.EqualFold(group.WhoCanViewMembership, "ALL_IN_DOMAIN_CAN_VIEW") || strings.EqualFold(group.WhoCanViewMembership, "ALL_MEMBERS_CAN_VIEW") || strings.EqualFold(group.WhoCanViewMembership, "ALL_MANAGERS_CAN_VIEW")) {
+			//return fmt.Errorf("wrong value specified for 'who_can_view_members' field (group: %s)", group.Name)
+			allTheErrors = append(allTheErrors, fmt.Errorf("wrong value specified for 'who_can_view_members' field (group: %s)", group.Name))
+		}
+
+		if !(strings.EqualFold(group.WhoCanApproveMembers, "ALL_MEMBERS_CAN_APPROVE") || strings.EqualFold(group.WhoCanApproveMembers, "ALL_MANAGERS_CAN_APPROVE") || strings.EqualFold(group.WhoCanApproveMembers, "ALL_OWNERS_CAN_APPROVE") || strings.EqualFold(group.WhoCanApproveMembers, "NONE_CAN_APPROVE")) {
+			//return fmt.Errorf("wrong value specified for 'who_can_approve_members' field (group: %s)", group.Name)
+			allTheErrors = append(allTheErrors, fmt.Errorf("wrong value specified for 'who_can_approve_members' field (group: %s)", group.Name))
+		}
+
+		if !(strings.EqualFold(group.WhoCanPostMessage, "NONE_CAN_POST") || strings.EqualFold(group.WhoCanPostMessage, "ALL_MANAGERS_CAN_POST") || strings.EqualFold(group.WhoCanPostMessage, "ALL_MEMBERS_CAN_POST") || strings.EqualFold(group.WhoCanPostMessage, "ALL_OWNERS_CAN_POST") || strings.EqualFold(group.WhoCanPostMessage, "ALL_IN_DOMAIN_CAN_POST") || strings.EqualFold(group.WhoCanPostMessage, "ANYONE_CAN_POST")) {
+			//return fmt.Errorf("wrong value specified for 'who_can_post' field (group: %s)", group.Name)
+			allTheErrors = append(allTheErrors, fmt.Errorf("wrong value specified for 'who_can_post' field (group: %s)", group.Name))
+		}
+
+		if !(strings.EqualFold(group.WhoCanJoin, "CAN_REQUEST_TO_JOIN") || strings.EqualFold(group.WhoCanJoin, "INVITED_CAN_JOIN") || strings.EqualFold(group.WhoCanJoin, "ALL_IN_DOMAIN_CAN_JOIN") || strings.EqualFold(group.WhoCanJoin, "ANYONE_CAN_JOIN")) {
+			//return fmt.Errorf("wrong value specified for 'who_can_contact_owner' field (group: %s)", group.Name)
+			allTheErrors = append(allTheErrors, fmt.Errorf("wrong value specified for 'who_can_contact_owner' field (group: %s)", group.Name))
+		}
+
+		if !(strings.EqualFold(group.AllowExternalMembers, "true") || strings.EqualFold(group.AllowExternalMembers, "false")) {
+			//return fmt.Errorf("wrong value specified for 'allow_external_members' field (group: %s)", group.Name)
+			allTheErrors = append(allTheErrors, fmt.Errorf("wrong value specified for 'allow_external_members' field (group: %s)", group.Name))
```

Comment:
All of this can disappear when we just specify the field as `bool` in our struct, instead of `string`.